src/components/menu/Aidrdrop/AirdropModal.tsx

```diff
@@ -0,0 +1,300 @@
+  return (
+    <>
+      <hr className="AirdropModal__space-line" style={style} />
+      <style jsx={true} global={true}>{`
+        .AirdropModal {
+          &__space-line {
+            width: 100%;
+            height: 1px;
+            border: 0;
+            border-radius: 9999px;
+            background: rgba(255, 255, 255, 0.2);
+          }
+        }
+      `}</style>
+    </>
+  );
+}
+
+function AirdropModal({ onRequestClose }: { onRequestClose: () => void }) {
+  const intl = useIntl();
+
+  const { currentAccount } = useUserWalletDataContext();
+  const [contract, setContract] = useState<ethers.Contract | null>(null);
+  const [claimableAmountRaw, setClaimableAmountRaw] = useState('0');
+  const [data, setData] = useState<{
+    totalAmount: string;
+    pendingAmount: string;
+    claimableAmount: string;
+  }>({
+    totalAmount: '-',
+    pendingAmount: '-',
+    claimableAmount: '-',
+  });
+  useEffect(() => {
+    const provider = new ethers.providers.Web3Provider((window as any).ethereum);
+    const signer = provider.getSigner();
+    const _contract = new ethers.Contract(CORN_AIRDROP_ADDRESS, AirdropAbi, signer);
+    setContract(_contract);
+    Promise.all([
+      _contract.getTotalAmount(currentAccount),
+      _contract.getPendingAmount(currentAccount),
+      _contract.getClaimableAmount(currentAccount),
+    ]).then(([_totalAmount, _pendingAmount, _claimableAmount]) => {
+      setClaimableAmountRaw(_claimableAmount.toString());
+      const totalAmount = valueToBigNumber(
+        normalize(valueToBigNumber(_totalAmount.toString()).toString(), CORN_DECIMALS)
+      ).toFixed(4, BigNumber.ROUND_DOWN);
+      const pendingAmount = valueToBigNumber(
+        normalize(valueToBigNumber(_pendingAmount.toString()).toString(), CORN_DECIMALS)
+      ).toFixed(4, BigNumber.ROUND_DOWN);
+      const claimableAmount = valueToBigNumber(
+        normalize(valueToBigNumber(_claimableAmount.toString()).toString(), CORN_DECIMALS)
+      ).toFixed(4, BigNumber.ROUND_DOWN);
+      setData({
+        totalAmount,
+        pendingAmount,
+        claimableAmount,
+      });
+      console.log({
+        _totalAmount: _totalAmount.toString(),
+        _pendingAmount: _pendingAmount.toString(),
+        _claimableAmount: _claimableAmount.toString(),
+      });
+    });
+  }, []);
+  return (
+    <ReactModal isOpen={true} overlayElement={_OverlayElement} contentElement={_ContentElement}>
+      <img
+        className="AirdropModal__close-icon"
+        src={CloseIcon}
+        alt="close"
+        onClick={onRequestClose}
+        role="button"
+      />
+      <div className="AirdropModal__title-section">
+        <img src={FireIcon} alt="fire" />
+        <h3>{intl.formatMessage(messages.airdropModalName)}</h3>
+      </div>
+      <SpaceLine style={{ margin: '16px 0' }} />
+      <div className="AirdropModal__CORN-balance">
+        <div>
+          <div className="corn-balance">1111.1111 CORN</div>
```

Comment:
The balance can be fetched with an ERC20 interface `balanceOf` of CORN contract.